api/proto/routing/v1/routing_service.proto

```diff
@@ -46,18 +46,42 @@ service RoutingService {
 }
 
 message PublishRequest {
-  // Reference to the record to be published.
-  string record_cid = 1;
+  oneof request {
+    // References to the records to be published.
+    RecordRefs record_refs = 1;
+
+    // Queries to match against the records to be published.
+    RecordQueries queries = 2;
+
+    // Publish all stored records.
+    bool all_records = 3;
+  }
```

Comment:
I'm not sure it's a good idea to mix the Publish function with the Search function. These are two separate things. I think Search is the client's job. The original 
`message PublishRequest {
  // Reference to the record to be published.
  string record_cid = 1;
}`
I think it would be appropriate if the long-stalled feature: streaming store client #240 https://github.com/agntcy/dir/pull/240 PR had been implemented. I think this would be the appropriate process
1. The client starts a Search and receives the CIDs
2. The client publishes the CIDs through the streaming client.

Reply:
@ramizpolic  OK. I understand. But we don't need this PublishRequest (refs, queries, all) whether we are on the server side or the client side. I think Publish has one task, which is to publish one or more records based on CID.